### PR TITLE
Release to PyPi, DockerHub, and GitHub

### PR DESCRIPTION
**Pull request by dblandin**
Add automated version releases to PyPi, DockerHub, and GitHub.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -68,9 +74,30 @@ publish-version:
...
+	twine upload dist/*
+
+publish-dockerhub:
+	docker tag hokusai:latest artsy/hokusai:$(VERSION)
+	docker push artsy/hokusai:$(VERSION)
+	docker tag hokusai:latest artsy/hokusai:latest
+	docker push artsy/hokusai:latest
+
+publish-github:
+	$(AWS) s3 cp \
```
Comment (dleve123) [inline Makefile]: This AWS cp copy block is used a couple of other times in this Makefile. Could it make sense / is it possible to extract another "Make function" and use that instead?

Comment (dleve123) [reply Makefile]: whoops I misread the order of the args here! carry on

File: Makefile [hunk 2]
```diff
@@ -68,9 +74,35 @@ publish-version:
 	  exit 1; \
 	fi
 
-tag:
-	$(GIT_TAG) --message v$(VERSION) v$(VERSION)
-	$(GIT_PUSH) origin refs/tags/v$(VERSION)
+publish-pip:
+	python setup.py sdist bdist_wheel
+	pip install twine
+	twine upload dist/*
```
Comment (izakp) [inline Makefile]: How are PyPi credentials set / injected?  Locally I have `~/.pypirc` set up

File: .circleci/config.yml [hunk 3]
```diff
@@ -63,6 +63,35 @@ jobs:
...
+      - run: make publish-dockerhub
+
+  release_github:
+    docker:
+      - image: golang:1.11
+    steps:
+      - checkout
+      - run: apt-get -qq update
+      - run: apt-get -qq install awscli
+      - run: go get -u github.com/tcnksm/ghr
```
Comment (izakp) [inline .circleci/config.yml]: Nice!

File: Makefile (hunk 2 above)
Comment (dblandin) [reply Makefile]: `TWINE_USERNAME` and `TWINE_PASSWORD` credentials are set in the environment via the Circle CI project configuration.

**Pull request merged**